code/go/0chain.net/miner/miner/main.go

```diff
@@ -204,7 +204,21 @@ func main() {
 	initIntegrationsTests(node.Self.Underlying().GetKey())
 
 	var server *http.Server
+
+	mux := http.NewServeMux()
+	http.DefaultServeMux = mux
+
 	if config.Development() {
+		if viper.GetBool("development.pprof") {
+			// start pprof server
+			go func() {
+				pprofMux := http.NewServeMux()
```

Comment:
we need a way to shut down it gracefully

Reply:
We'll need to add more invasive changes to the current code then. because we have to shutdown both servers following the same signal.

Reply:
Added code to shutdown prof server. Works with test code, please take a look yourself.